### PR TITLE
refactor!: rename `StakePoolSearchProvider` to `StakePoolProvider`

### PR DESCRIPTION
**Pull request by lgobbi-atix**
# Context

Generalize `StakePoolSearchProvider` as part of the refactor to remove `WalletProvider`

# Proposed Solution

Rename `StakePoolSearchProvider` to `StakePoolProvider` and move `stakePoolStats` from `WalletProvider`

# Important Changes Introduced
- Renamed all occurrences of `StakePoolSearchProvider` to `StakePoolProvider`
- Removed `stakePoolStats` from `WalletProvider`
- Added `stakePoolStats` to `StakePoolProvider` with its corresponding query and endpoint

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
I found a few more occurrences of "StakePoolSearch":

- [This](https://github.com/input-output-hk/cardano-js-sdk/blob/e8142e90997ae2e1a31ea2337d527dbe3fde85b5/packages/wallet/test/services/ProviderTracker/TrackedUtxoProvider.test.ts#L6) is an error, but please fix in this PR too.
- https://github.com/input-output-hk/cardano-js-sdk/blob/773c926b78ee9bfbad7e25e54288401298a3f548/packages/cardano-services/test/StakePool/StakePoolHttpService.test.ts#L85
- https://github.com/input-output-hk/cardano-js-sdk/blob/773c926b78ee9bfbad7e25e54288401298a3f548/packages/cardano-services/test/StakePool/StakePoolHttpService.test.ts#L110

**Review by rhyslbw — APPROVED** [review]
Thanks @lgobbi-atix

File: packages/blockfrost/src/blockfrostWalletProvider.ts [hunk 1]
```diff
@@ -50,26 +50,18 @@ export const blockfrostWalletProvider = (blockfrost: BlockFrostAPI, logger = dum
...
-      return newCount;
-    };
-    return {
-      qty: {
-        active: await tallyPools('pools'),
-        retired: await tallyPools('poolsRetired'),
-        retiring: await tallyPools('poolsRetiring')
-      }
-    };
+  const utxoByAddresses: WalletProvider['utxoByAddresses'] = async (addresses) => {
```
Comment (rhyslbw) [inline packages/blockfrost/src/blockfrostWalletProvider.ts:53]: Why is this being reintroduced? 
https://github.com/input-output-hk/cardano-js-sdk/pull/240/commits/e8142e90997ae2e1a31ea2337d527dbe3fde85b5

Comment (lgobbi-atix) [reply packages/blockfrost/src/blockfrostWalletProvider.ts]: Removed

**Review by mkazlauskas — APPROVED** [review]
Looks good to me!

**Review by rhyslbw — DISMISSED** [review]
``` ts
export const mockWalletProvider = (): WalletProvider => ({
  ...
});
```

Looks like we're missing the typing in both ` mockAssetProvider` and `mockNetworkInfoProvider` too. They can all be committed along with the fix.

**Comment by mkazlauskas** [comment]
> Looks like we're missing the typing in both mockAssetProvider and mockNetworkInfoProvider too. They can all be committed along with the fix.

That is actually intentionally not typed, so that tests can do stuff like `provider.method.mockResolvedValue(...)` instead of `(provider as jest.Mock).method.mockResolvedValue(...)`.

It's losing type-checks for compatibility with provider interface, but is more convenient to use. I was thinking we can rely on tests to fail if some method is missing or is incompatible.

**Comment by rhyslbw** [comment]
> I was thinking we can rely on tests to fail if some method is missing or is incompatible.

The only gap here is a superset of methods, but I don't think it's worth getting hung up in this now.

**Review by rhyslbw — APPROVED** [review]
Thanks @lgobbi-atix

**Pull request merged**